presto-geospatial-toolkit/src/test/java/com/facebook/presto/geospatial/rtree/TestFlatbush.java

```diff
@@ -0,0 +1,300 @@
+    private static final OGCGeometry POINT_Z = new OGCPoint(new Point(6.0, 6.0), null);
+    private static final OGCGeometry POINT_W = new OGCPoint(new Point(20.0, 20.0), null);
+
+    private static <T extends HasExtent> List<T> findIntersections(Flatbush<T> rtree, Rectangle rectangle)
+    {
+        List<T> results = new ArrayList<>();
+        rtree.findIntersections(rectangle, results::add);
+        return results;
+    }
+
+    /*
+     * Asserts the two lists of Rectangles are equal after sorting.
+     */
+    private static <T> void assertEqualsSorted(List<T> actual, List<T> expected, Comparator<T> comparator)
+    {
+        List<T> actualSorted = actual.stream().sorted(comparator).collect(toImmutableList());
+        List<T> expectedSorted = expected.stream().sorted(comparator).collect(toImmutableList());
+
+        assertEquals(actualSorted, expectedSorted);
+    }
+
+    @Test
+    public void testEmptyFlatbush()
+    {
+        Flatbush<Rectangle> rtree = new Flatbush<>(new Rectangle[] {});
+        assertEquals(findIntersections(rtree, EVERYTHING), ImmutableList.of());
+    }
+
+    @Test
+    public void testSingletonFlatbush()
+    {
+        List<Rectangle> items = new ArrayList<>();
+        items.add(new Rectangle(0, 0, 1, 1));
+        Flatbush<Rectangle> rtree = new Flatbush<>(items.toArray(new Rectangle[] {}));
+
+        assertEquals(findIntersections(rtree, EVERYTHING), items);
+        // hit
+        assertEquals(findIntersections(rtree, new Rectangle(1, 1, 2, 2)), items);
+        // miss
+        assertEquals(findIntersections(rtree, new Rectangle(-1, -1, -0.1, -0.1)), ImmutableList.of());
+    }
+
+    @Test
+    public void testSingletonFlatbushXY()
+    {
+        // Because mixing up x and y is easy to do...
+        List<Rectangle> items = new ArrayList<>();
+        items.add(new Rectangle(0, 10, 1, 11));
+        Flatbush<Rectangle> rtree = new Flatbush<>(items.toArray(new Rectangle[] {}));
+
+        // hit
+        assertEquals(findIntersections(rtree, new Rectangle(1, 11, 2, 12)), items);
+        // miss
+        assertEquals(findIntersections(rtree, new Rectangle(11, 1, 12, 2)), ImmutableList.of());
+    }
+
+    @Test
+    public void testDoubletonFlatbush()
+    {
+        // This is the smallest Rtree with height > 1
+        // Also test for some degeneracies
+        List<Rectangle> items = new ArrayList<>();
+        Rectangle rect0 = new Rectangle(1, 1, 1, 1);
+        items.add(rect0);
+        Rectangle rect1 = new Rectangle(-1, -2, -1, -1);
+        items.add(rect1);
+        items.sort(RECTANGLE_COMPARATOR);
+
+        Flatbush<Rectangle> rtree = new Flatbush<>(items.toArray(new Rectangle[] {}));
+
+        List<Rectangle> allResults = findIntersections(rtree, EVERYTHING);
+        allResults.sort(RECTANGLE_COMPARATOR);
+        assertEquals(allResults, items);
+
+        Rectangle hit0 = new Rectangle(1, 1, 2, 2);
```

Comment:
avoid a1, a2, ... variable names; it seems like inlining variables here makes it easier to read

```
        assertEquals(findIntersections(rtree, new Rectangle(1, 1, 2, 2)), ImmutableList.of(rect0));
        assertEquals(findIntersections(rtree, new Rectangle(-2, -2, -1, -2)), ImmutableList.of(rect1));
        // This should test missing at the root level
        assertEquals(findIntersections(rtree, new Rectangle(10, 10, 12, 12)), ImmutableList.of());
        // This should test missing at the leaf level
        assertEquals(findIntersections(rtree, new Rectangle(0, 0, 0, 0)), ImmutableList.of());
```

ditto testTwoLevelFlatbush